cabot/cabotapp/models.py

```diff
@@ -1146,6 +1173,146 @@ def __unicode__(self):
         return "%s: %s to %s%s" % (self.user.username, self.start, self.end, deleted)
 
 
+class ResultFilter(models.Model):
+    """Filter that matches StatusCheckResults."""
+    class Meta:
+        abstract = True
+
+    status_check = models.ForeignKey(StatusCheck, null=False)
+
+    MATCH_CHECK = 'C'
+    MATCH_TYPE_CHOICES = (
+        (MATCH_CHECK, 'Match check.'),
+    )
+    match_if = models.TextField(max_length=1, null=False, blank=False, default=MATCH_CHECK, choices=MATCH_TYPE_CHOICES)
+
+    def matches_result(self, result):
+        # type: (StatusCheckResult) -> bool
+
+        # status_check must match, regardless of match type
+        if result.status_check_id != self.status_check_id:
+            return False
+
+        if self.match_if == self.MATCH_CHECK:
+            return True
+
+        raise NotImplementedError()
```

Comment:
why `NotImplementedError` not `return False`?

Reply:
`match_if` acts like an enum, and I want all possible cases of the enum to be handled. This way, if someone adds a new value to the enum but forgets to update this, they'll get an exception instead of their ack mysteriously not activating.